src/cargo/util/toml.rs

```diff
@@ -741,6 +745,18 @@ fn unique_names_in_targets(targets: &[TomlTarget]) -> Result<(), String> {
     Ok(())
 }
 
+/// Will check a list of build targets, and make sure the target names are unique within a vector.
+/// If not, the name of the offending build target is returned.
+fn unique_build_targets(targets: &Vec<Target>) -> Result<(), Display> {
+    let mut seen = HashSet::new();
+    for v in targets.iter().map(|e| e.src_path()) {
+        if !seen.insert(v.clone()) {
+            return Err(v.display());
```

Comment:
It should be fine here to just return a `String` instead of importing `path::Display` as well.